weldx/core.py

```diff
@@ -708,3 +711,557 @@ def units(self) -> pint.Unit:
+            # evaluate expression
+            coords = self._call_preprocess_coords(**kwargs)
+            coords_unit_adj = {k: v.pint.dequantify() for k, v in coords.items()}
+            data = self._obj.evaluate(**coords).assign_coords(coords_unit_adj)
+            # todo use func of CFabry to turn coord attrs into units
+            return type(self)(data)
+        else:
+            # turn passed coords into parameters of the expression
+            new_series = deepcopy(self)
+            for k, v in kwargs.items():
+                new_series._data.set_parameter(k, (v, self._symbol_dims[k]))
+                new_series._symbol_dims.pop(k)
+                new_series._variable_units.pop(k)
+            return new_series
+
+    def __call__(self, **kwargs) -> GenericSeries:
+        """Evaluate the generic series at discrete coordinates.
+
+        Parameters
+        ----------
+        kwargs:
+            An arbitrary number of keyword arguments. The key must be a dimension name
+            of the `GenericSeries` and the values are the corresponding coordinates
+            where the `GenericSeries` should be evaluated.
+
+        Returns
+        -------
+        GenericSeries :
+            A new generic series containing the discrete values at the desired
+            coordinates.
+
+        """
+        if self.is_expression:
+            return self._call_expr(**kwargs)
+
+        coords = self._call_preprocess_coords(**kwargs)
+        for k in coords.keys():
+            if k not in self._obj.dims:
+                raise KeyError(f"'{k}' is not a valid dimension.")
+        return type(self)(ut.xr_interp_like(self._obj, coords))
+
+    def __getitem__(self, *args):
+        """Get a subset of a discrete `GenericSeries` by indices."""
+        if isinstance(self._obj, xr.DataArray):
+            return self._obj.__getitem__(*args)
+        return NotImplementedError
+
+    def interp_like(
+        self, obj: Any, dimensions: List[str] = None, accessor_mappings: Dict = None
+    ) -> GenericSeries:
+        """Interpolate using the coordinates of another object.
+
+        Parameters
+        ----------
+        obj :
+            An object that provides the coordinate values.
+        dimensions :
+            The dimensions that should be interpolated. If `None` is passed, all
+            dimensions will be interpolated
+        accessor_mappings :
+            A mapping between the dimensions of the generic series and the corresponding
+            coordinate accessor of the provided object. This can be used if the
+            coordinate names do not match for the time series and the provided object.
+
+        Returns
+        -------
+        GenericSeries :
+            A new generic series containing discrete values for the interpolated
+            dimensions.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def coordinates(self) -> Union[None, pint.Quantity, Dict[str, pint.Quantity]]:
```

Comment:
I suggest outputting a similar format to the xarray coords object (so `Dict[str, pint.Quantity]`